### PR TITLE
Logic for UserRegistration.

### PR DESCRIPTION
**Pull request by qvalentin**
Just so i can already comment on it.

### CODE REVIEW

File: src/main/java/de/filefighter/rest/domain/user/business/UserBusinessService.java [hunk 1]
```diff
@@ -74,4 +70,54 @@ public User findUserByUsername(String username) {
...
+
+        if (null != user)
+            throw new UserAlreadyExistsException("Username already taken.");
+
+        // check pws.
+        String password = newUser.getPassword();
+        passwordIsValid(password);
+
+        String confirmationPassword = newUser.getConfirmationPassword();
+        passwordIsValid(confirmationPassword);
```
Comment (qvalentin) [inline src/main/java/de/filefighter/rest/domain/user/business/UserBusinessService.java:92]: not necessary. If the comparision is not succesful the error message will be more precise

File: src/main/java/de/filefighter/rest/domain/user/business/UserBusinessService.java [hunk 2]
```diff
@@ -74,4 +70,54 @@ public User findUserByUsername(String username) {
...
+            }
+        }
+
+        //create new user.
+        userRepository.save(UserEntity.builder()
+                .lowercaseUsername(username.toLowerCase())
+                .username(username)
+                .password(password)
+                .refreshToken(AccessTokenBusinessService.generateRandomTokenValue())
+                .userId(getUserCount() + 1)
```
Comment (qvalentin) [inline src/main/java/de/filefighter/rest/domain/user/business/UserBusinessService.java:116]: counting up is not perfect, as user Id coud be easily guessed, but could be ok here

**Review by qvalentin — COMMENTED** [review]
just some thoughts I had

File: src/main/java/de/filefighter/rest/domain/user/business/UserAuthorizationService.java [hunk 3]
```diff
@@ -32,13 +28,13 @@ public UserAuthorizationService(UserRepository userRepository, UserDtoService us
     }
 
     public User authenticateUserWithUsernameAndPassword(String base64encodedUserAndPassword) {
-        String decodedUsernameUndPassword;
+        String decodedUsernameUndPassword = "";
```
Comment (qvalentin) [inline src/main/java/de/filefighter/rest/domain/user/business/UserAuthorizationService.java:31]: Und ?

Comment (open-schnick) [reply src/main/java/de/filefighter/rest/domain/user/business/UserAuthorizationService.java:31]: needs to be initialized

File: src/main/java/de/filefighter/rest/domain/user/business/UserBusinessService.java (hunk 2 above)
Comment (open-schnick) [reply src/main/java/de/filefighter/rest/domain/user/business/UserBusinessService.java:130]: we will change the ids anyway i guess.

File: pom.xml [hunk 4]
```diff
@@ -49,6 +49,11 @@
             <version>1.18.16</version>
         </dependency>
 
+        <dependency>
+            <groupId>org.apache.httpcomponents</groupId>
+            <artifactId>httpclient</artifactId>
+        </dependency>
```
Comment (open-schnick) [inline pom.xml:55]: fixing a bug in restTemplate. Overwriting the default httpRequestFactory.

**Review by Gimleux — APPROVED** [review]

**Pull request merged**